app/models/manageiq/providers/openstack/infra_manager/refresh_parser.rb

```diff
@@ -150,12 +151,16 @@ def load_hosts
         end
       end
 
-      def get_extra_host_attributes(host)
-        extra_attrs = @data_index.fetch_path(:cloud_object_store_objects, 'extra_hardware-' + host.uuid, :content)
-        return {} if extra_attrs.blank?
-        # Convert list of tuples from Ironic extra to hash. E.g. [[a1, a2, a3, a4], [a1, a2, b3, b4], ..] converts to
-        # {a1 => {a2 => {a3 => a4, b3 => b4}}}, so we get constant access to sub indexes.
-        JSON.parse(extra_attrs).each_with_object({}) { |attr, obj| ((obj[attr[0]] ||= {})[attr[1]] ||= {})[attr[2]] = attr[3] if attr.count >= 4 }
+      def get_introspection_details(host)
+        intro_details = @introspection_service.get_introspection_details(host.uuid).body unless
+          @introspection_service.nil?
+        return Hash.new if intro_details.blank?
+        intro_details
```

Comment:
I keep getting caught up in some of the checks here...

I think this could be:

```ruby
return {} unless @introspection_service
@introspection_service.get_introspection_details(host.uuid).body
```

Then, if there's no `@introspection_service`, it returns an empty hash.  If the `@introspection_service` call yields an empty hash, it returns an empty hash.  Otherwise, it returns the body of the call.

I don't think that the `Excon` response bodies can be `nil`.  But, if they can, then this could be:

```ruby
return {} unless @introspection_service
@introspection_service.get_introspection_details(host.uuid).body || {}
```